File: fact.cpp
#include <iostream>
using namespace std;

int fact(int n) {
	if (n <= 1) {
		return n;
	}
	return n * fact(n - 1);
}
int main() {
	int num = 7;
	cout << "Result: " << fact(num);
	return 0;
}

File: fact.rs
fn fact(n: u64) -> u64 {
	match n {
		0 | 1 => 1,
		_ => fact(n - 1) * n,
	}
}

fn main() {
	let num: u64 = 7;
	println!("{}", fact(num));
}
